osmUtils/osmDownload.py

```diff
@@ -0,0 +1,60 @@
+from .utils_osm import generate_filter, retrieve_osm, generate_osm_gdf
+from .settings import DEFAULT_PATH, DEFAULT_TIMEOUT, DEFAULT_OVERPASS_ENDPOINT
+
+class OsmDownload:
+    """
+    Download OSM ways and nodes within a given geometry from the Overpass API.
+    
+    Parameters
+    ----------
+    geometry: shapely.geometry.Polygon or shapely.geometry.MultiPolygon
+        geographic boundaries to fetch geometries within
+    osm_type: string
+        type of filter to retieve if custom_filter is None (e.g 'all_roads')
+    infrastructure: string
+        infrastructure type that will be use to build the overpas api query (e.g. 'way["highway"]')
+    custom_filter: string
+        a custom filter to be used instead of the already defined in the osm_type
+        
+    Returns
+    -------
+    osmData: geopandas.GeoDataFrame
+        response retrieved from overpass api in a geopandas.GeoDataFrame
+    
+    """
+    def __init__(self, geometry,  osm_type, infrastructure, custom_filter=None, output_path=None):
+
+        self.geometry = geometry
+        self.infrastructure = infrastructure
```

Comment:
Should we have fallbacks? Users can access this class directly, remember.

Reply:
i have add a fallback for infrastructure and osm type. I guess geometry should always be mandatory.